docs/preview/duckdb/backups_and_recovery.md

```diff
@@ -0,0 +1,111 @@
+- [Continuous Archiving and Point-in-Time Recovery (PITR)](https://www.postgresql.org/docs/current/continuous-archiving.html): This is a more complex approach but allows recovery to a specific point in time. For DuckLake, this means you can recover to a specific snapshot without losing any transactions.
+
+Nothe that the SQL dump approach can also be managed by DuckDB using the [Postgres extension](https://duckdb.org/docs/stable/core_extensions/postgres). In fact, the backup can be a DuckDB file.
+
+```sql
+-- Backup
+ATTACH 'postgres:connection_string' AS main (READ_ONLY);
+ATTACH 'duckdb:backup.db' AS backup;
+COPY DATABASE FROM main TO backup;
+
+-- Recover
+ATTACH 'postgres:connection_string' AS main;
+ATTACH 'duckdb:backup.db' AS backup (READ_ONLY);
+COPY DATABASE FROM backup TO main;
+ATTACH 'ducklake:postgres:connection_string' AS my_ducklake;
+```
+
+> Cloud-hosted PostgreSQL solutions may offer different mechanisms. We encourage you to check what your specific vendor recommends as a strategy for backup and recovery.
+
+## Storage Backup and Recovery
+
+Backup and recovery of the data files also depend on the storage you are using. In this document, we will only focus on cloud-based object storage since it is the most common for Lakehouse architectures.
+
+### S3
+
+In S3, there are three main mechanisms that AWS offers to back up and/or restore data:
+
+- [Cross-bucket replication](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html)
+- [S3 backup service](https://docs.aws.amazon.com/aws-backup/latest/devguide/s3-backups.html)
+- [Enable S3 versioning](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html)
+
+Both the S3 backup service and S3 object versioning will restore data files in the same bucket. On the other hand, cross-bucket replication will copy the files to a different bucket, and therefore your DuckLake initialization should change:
+
+```sql
+-- Before
+ATTACH 'ducklake:some.db' AS my_ducklake (DATA_PATH 's3://og-bucket/')
+
+-- After
+ATTACH 'ducklake:some.db' AS my_ducklake (DATA_PATH 's3://replication-bucket/')
+```
+
+### GCS
+
+GCS has similar mechanisms to back up and/or restore data:
+
+- [Cross-bucket replication](https://cloud.google.com/storage/docs/using-cross-bucket-replication)
+- [Backup and DR service](https://cloud.google.com/backup-disaster-recovery/docs/concepts/backup-dr)
+- [Object versioning](https://cloud.google.com/storage/docs/object-versioning) with soft deletes enabled
+
+Regarding cross-bucket replication, repointing to the new bucket will be necessary.
+
+```sql
+-- Before
+ATTACH 'ducklake:some.db' AS my_ducklake (DATA_PATH 'gs://og-bucket/')
+
+-- After
+ATTACH 'ducklake:some.db' AS my_ducklake (DATA_PATH 'gs://replication-bucket/')
+```
```

Comment:
Maybe it's worth saying that backups will keep working AS LONG AS you don't cleanup file they reference.

Basically, if you use https://ducklake.select/docs/stable/duckdb/maintenance/cleanup_old_files.html, idea should be like:

```
1. create ducklake
2. inserts / update / what not
3. create backup1 (time T=3)
4. more inserts / update / what not
5. merge files / cleanup orphaned files (with older timestamp than T=3 !!!), backup 1 still valid
6. create backup2 (time T=6)
7. more inserts / update / what not
8. merge files / cleanup orphaned files (with timestamp just older than T=6 !!!), NOW backup1 is invalid, !!
9. create backup3 (time T=9)
....
```

Reply:
Unsure if this makes exact sense, @pdet ???

Basically, at some point backups will reference non-existing files, unsure what are the exact conditions, but I think it's handy to say explcitly.

Reply:
Yeah I agree I think the idea is: don't use cleanup_old_files after a backup, always before a backup.